File: backend/internal/service/menu.go
package service

import (
	"backend/internal/model"
	"backend/internal/repository"
)

// MenuService 菜单服务
type MenuService struct {
	MenuRepo *repository.MenuRepository
}

// NewMenuService 创建菜单服务
func NewMenuService(menuRepo *repository.MenuRepository) *MenuService {
	return &MenuService{
		MenuRepo: menuRepo,
	}
}

// CreateMenu 创建菜单
func (s *MenuService) CreateMenu(menu *model.Menu) error {
	return s.MenuRepo.Create(menu)
}

// GetMenuByID 根据ID获取菜单
func (s *MenuService) GetMenuByID(id uint) (*model.Menu, error) {
	return s.MenuRepo.GetByID(id)
}

// GetMenuTree 获取菜单树
func (s *MenuService) GetMenuTree(parentID *uint) ([]model.Menu, error) {
	return s.MenuRepo.GetByParentID(parentID)
}

// GetAllMenus 获取所有菜单
func (s *MenuService) GetAllMenus() ([]model.Menu, error) {
	return s.MenuRepo.GetAll()
}

// UpdateMenu 更新菜单
func (s *MenuService) UpdateMenu(menu *model.Menu) error {
	return s.MenuRepo.Update(menu)
}

// DeleteMenu 删除菜单
func (s *MenuService) DeleteMenu(id uint) error {
	return s.MenuRepo.Delete(id)
}

// GetUserMenus 获取当前用户的菜单
func (s *MenuService) GetUserMenus(userID uint) ([]model.Menu, error) {
	return s.MenuRepo.GetByUserID(userID)
}

File: frontend/src/api/menu.ts
// api/menu.ts
import request from './request'

// 创建菜单
export const createMenu = (data: any) => {
  return request.post('/menus', data)
}

// 获取菜单树（用于菜单管理）
export const getMenuTree = (parentId?: number) => {
  const params: any = {}
  if (parentId !== undefined) {
    params.parent_id = parentId
  }
  return request.get('/menus/tree', { params })
}

// 获取所有菜单
export const getAllMenus = () => {
  return request.get('/menus/all')
}

// 获取菜单详情
export const getMenu = (id: number) => {
  return request.get(`/menus/${id}`)
}

// 更新菜单
export const updateMenu = (id: number, data: any) => {
  return request.put(`/menus/${id}`, data)
}

// 删除菜单
export const deleteMenu = (id: number) => {
  return request.delete(`/menus/${id}`)
}

// 获取当前用户的菜单（用于动态路由）
export const getUserMenus = () => {
  return request.get('/menus')
}